devito/types/basic.py

```diff
@@ -1315,7 +1321,7 @@ class LocalObject(AbstractObject):
 # - To override SymPy caching behaviour
 
 
-class IndexedData(sympy.IndexedBase, Pickable):
+class IndexedData(sympy.IndexedBase, Pickable, CodeSymbol):
```

Comment:
the changes here are to pass a Function by "data pointer" rather by "struct dataobj *", ie it allows the generation of

```
void foo(float * f)
```

besides

```
void foo(struct dataobj * f_vec)
```